### PR TITLE
samples: matter: Updated Matter to pull BLE related improvements

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Updated Matter version to pull cnfiguration improvements that:
* Enable Wi-Fi/Bluetooth LE coexistance
* Enable erasing settings partition during DFU over SMP
* Change Bluetooth LE connection parameters to make connections with Matter controllers work in a more reliable way.

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

### CODE REVIEW

**Review by Damian-Nordic — APPROVED** [review]
I guess changing the parameters requires release notes.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -82,13 +82,16 @@ Matter
   * Default support for nRF7002 revision B.
   * Specific QR code and onboarding information in the documentation for each :ref:`Matter sample <matter_samples>` and the :ref:`Matter weather station <matter_weather_station_app>`.
   * The Bluetooth LE advertising arbiter class that enables easier coexistence of application components that want to advertise their Bluetooth LE services.
+  * Support for erasing settings partition during DFU over Bluetooth LE SMP for the nRF52 family.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:85]:
```suggestion
  * Support for erasing settings partition during DFU over Bluetooth LE SMP for the Nordic nRF52 Series' SoCs.
```
As per https://projecttools.nordicsemi.no/confluence/display/MAR/Product+names

**Review by greg-fer — CHANGES_REQUESTED** [review]

**Review by ArekBalysNordic — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -82,13 +82,16 @@ Matter
...
+  * Support for erasing settings partition during DFU over Bluetooth LE SMP for the Nordic nRF52 Series' SoCs.
+  * Enabled Wi-Fi and Bluetooth LE coexistence.
 
 * Updated:
 
   * The default heap implementation to use Zephyr's ``sys_heap`` (:kconfig:option:`CONFIG_CHIP_MALLOC_SYS_HEAP`) to better control the RAM usage of Matter applications.
   * :ref:`ug_matter_device_certification` page with a section about certification document templates.
   * :ref:`ug_matter_overview_commissioning` page with information about :ref:`ug_matter_network_topologies_commissioning_onboarding_formats`.
   * Default retry intervals used by Matter Reliability Protocol for Matter over Thread to account for longer round-trip times in Thread networks with multiple intermediate nodes.
+  * Bluetooth LE connection timeout and parameters update timeout to make communication over Bluetooth LE make more reliably.
```
Comment (wiba-nordic) [inline doc/nrf/releases/release-notes-changelog.rst:94]:
```suggestion
  * Bluetooth LE connection timeout and parameters update timeout to make communication over Bluetooth LE more reliable.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

**Review by wiba-nordic — APPROVED** [review]
I am also approving in greg-fer's stead since he is out of office.

File: doc/nrf/releases/release-notes-changelog.rst (hunk 1 above)
Comment (wiba-nordic) [reply doc/nrf/releases/release-notes-changelog.rst:89]: Did this change get reverted?

Comment (kkasperczyk-no) [reply doc/nrf/releases/release-notes-changelog.rst]: Yes, that was unintentional problem during rebase. It's fixed now.

**Pull request merged**